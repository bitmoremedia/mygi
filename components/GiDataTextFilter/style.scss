@import '../../css/variables';

.gi-data-text-filter {
  position: relative;
  padding-top: 10px;
}
form.gi-data-text-filter__form {
  margin: 0px;
}
input.gi-data-text-filter__input {
  padding-right: 115px;
}
.gi-data-text-filter__clear-button {
  position: absolute;
  right: 0px;
  bottom: 5px;
  border-top-left-radius: 0px;
  border-bottom-left-radius: 0px;
}
.gi-data-text-filter__help-text {
  position: absolute;
  left: 5px;
  bottom: -5px;
  font-size: 12px;
  color: $main;
}
