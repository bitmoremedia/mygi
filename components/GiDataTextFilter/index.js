import React, { Component, PropTypes } from 'react'

import './style.scss'

class GiDataTextFilter extends Component {

  constructor(props, context) {
    super(props, context)
    this.state = {
      showHelp: false,
    }
    this.textFilterInput = undefined
  }

  onFocus = () => {
    this.setState({ showHelp: true })
  }

  onBlur = () => {
    this.setState({ showHelp: false })
  }

  handleSubmit = (event) => {
    event.preventDefault()
    this.textFilterInput.blur()
    this.onBlur()
  }

  handleChange = (event) => {
    this.props.setTextFilter(event.target.value)
  }

  clearFilter = () => {
    this.props.setTextFilter('')
  }

  render() {
    const { onFocus, onBlur, handleChange, handleSubmit, clearFilter } = this
    const { textFilter } = this.props
    const { showHelp } = this.state
    const placeholderText = 'Filter'
    const clearButton = (
      <button
        className="gi-data-text-filter__clear-button"
        onClick={clearFilter}
      >
       Clear
     </button>)
    const helpText = (
      <div className="gi-data-text-filter__help-text">
        {"Filter on 'Text' or a GI value range e.g. 1-30"}
      </div>)
    return (
      <div className="gi-data-text-filter">
        <form
          className="gi-data-text-filter__form"
          onSubmit={handleSubmit}
          autoComplete="off"
        >
          <input
            className="gi-data-text-filter__input"
            type="text"
            autoComplete="off"
            ref={(c) => { this.textFilterInput = c }}
            name="text-filter"
            placeholder={placeholderText}
            value={textFilter}
            onChange={handleChange}
            onFocus={onFocus}
            onBlur={onBlur}
          />
        </form>
        {
          textFilter && clearButton
        }
        {
          showHelp && helpText
        }
      </div>
    )
  }
}

GiDataTextFilter.propTypes = {
  setTextFilter: PropTypes.func.isRequired,
  textFilter: PropTypes.string.isRequired,
}

export default GiDataTextFilter
