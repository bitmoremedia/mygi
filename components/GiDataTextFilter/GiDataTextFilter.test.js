import React from 'react'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'

import GiDataTextFilter from './index'

test('component renders without errors', () => {
  const wrapper = mount(
    <GiDataTextFilter
      setTextFilter={() => {}}
      textFilter=""
    />,
  )
  expect(wrapper.exists()).toBe(true)
})

test('component output as expected', () => {
  const component = renderer.create(
    <GiDataTextFilter
      setTextFilter={() => {}}
      textFilter=""
    />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('setTextFilter is called when text is changed in the input form field', () => {
  const setTextFilter = jest.fn()
  const wrapper = shallow(
    <GiDataTextFilter
      setTextFilter={setTextFilter}
      textFilter=""
    />,
  )
  wrapper.find('input').simulate('change', { target: { value: 'Some Text' } })
  expect(setTextFilter).toHaveBeenCalledWith('Some Text')
})

test('setTextFilter is called when clear button is pressed, setting the new value as blank', () => {
  const setTextFilter = jest.fn()
  const wrapper = shallow(
    <GiDataTextFilter
      setTextFilter={setTextFilter}
      textFilter="Initial Text"
    />,
  )
  wrapper.find('button').simulate('click')
  expect(setTextFilter).toHaveBeenCalledWith('')
})

test('help text is displayed ONLY when the user has focus on the text input form field', () => {
  const wrapper = shallow(
    <GiDataTextFilter
      setTextFilter={() => {}}
      textFilter=""
    />,
  )
  expect(wrapper.instance().state.showHelp).toBe(false)
  wrapper.find('input').simulate('focus')
  expect(wrapper.instance().state.showHelp).toBe(true)
  wrapper.find('input').simulate('blur')
  expect(wrapper.instance().state.showHelp).toBe(false)
})

test('when the user hits return the text input loses focus and any help is hidden', () => {
  const wrapper = mount(
    <GiDataTextFilter
      setTextFilter={() => {}}
      textFilter="Initial Text"
    />,
  )
  expect(wrapper.instance().state.showHelp).toBe(false)
  wrapper.find('input').simulate('focus')
  expect(wrapper.instance().state.showHelp).toBe(true)
  wrapper.find('form').simulate('submit')
  expect(wrapper.instance().state.showHelp).toBe(false)
})
