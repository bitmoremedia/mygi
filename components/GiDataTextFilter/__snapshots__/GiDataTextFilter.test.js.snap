// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component output as expected 1`] = `
<div
  className="gi-data-text-filter"
>
  <form
    autoComplete="off"
    className="gi-data-text-filter__form"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="gi-data-text-filter__input"
      name="text-filter"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Filter"
      type="text"
      value=""
    />
  </form>
  
</div>
`;
